app/uk/gov/hmrc/gform/playcomponents/RequestHeaderService.scala

```diff
@@ -39,7 +40,7 @@ final class RequestHeaderService(
     formTemplateIdParamIndex match {
       case Some(i) if i =!= -1 =>
         val templateId = rh.uri.split("\\?")(0).split("/")(i)
-        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(rh, rh.session)
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(rh)
```

Comment:
This actually make sense. I bet session is not properly initiated at this point, so `fromRequestAndSession` ended up with failed cookie. 

Can we undo other changes and make this single line a fix?